### ISSUE TITLE
CoreOS Support

### ISSUE THREAD

**Issue by pkudaiyar** [intelsdi-x/snap-plugin-collector-docker#33]
I am trying to install snap and docker plugin in the CoreOS cluster but it is not working. Is Snap currently supported for CoreOS operating system?

**Comment by mjbrender** [comment]
I've had Snap running on CoreOS before. Glad to help if I can. 

What are you running into? CoreOS requires Snap to be in a container and depending on what plugins you're using we could share a Dockerfile that'll get you there. cc @thomastaylor312 @andrzej-k

**Comment by pkudaiyar** [comment]
I wanted to collect docker metrics from the CoreOS system. Yes, I did installed dockerized snap.

**Comment by pkudaiyar** [comment]
I built my docker images using the docker scripts from below link
https://github.com/intelsdi-x/snap-plugin-collector-docker/blob/master/scripts/Dockerfile

**Comment by IzabellaRaulin** [comment]
Hello @pkudaiyar, could You also put command how you run docker ?

**Comment by pkudaiyar** [comment]
docker run -d -v /proc:/hostproc -v /sys/fs/cgroup/:/sys/fs/cgroup -v /usr/bin/docker:/usr/bin/docker -v /var/run/docker.sock:/var/run/docker.sock <my-image-name> bash

**Comment by IzabellaRaulin** [comment]
Thanks, Could you give more details what happens:
- are you able to load docker collector plugin `snapctl plugin load snap-plugin-collector-docker`
- are you able to see exposed metrics `snapctl metric list`

Coud you also put your task manifest?

**Comment by pkudaiyar** [comment]
Error creating task:
Metric not found: /intel/linux/docker/*/blkio_stats/io_serviced_recursive/0/value
Metric not found: /intel/linux/docker/*/cpu_stats/cpu_usage/total_usage
Metric not found: /intel/linux/docker/*/memory_stats/usage/usage

my task manifest file is based on the below link except i replaced the container id with '*' to collect metrics from all the running containers.
https://github.com/intelsdi-x/snap-plugin-collector-docker/blob/master/examples/tasks/docker-file.json

**Comment by IzabellaRaulin** [comment]
Would you mind try this one: https://github.com/IzabellaRaulin/snap-plugin-collector-docker/tree/migration_from_kubesnap (it still WIP, but I am just finishing unit tests)?. For this version add to your Dockerfile the mountpoint of procfs (defaults to `/proc`):
```
ENV PROCFS_MOUNT /proc_host
```

Please, let's me know if this works on CoreOs.

**Comment by nanliu** [comment]
I've ran into issues with runc/libcontainer detection of the specific filepath under /sys/fs/cgroup/ correctly. Since the metrics failing to load are all in:

```
/sys/fs/cgroup/blkio/docker/{uuid}
/sys/fs/cgroup/cpuacct/docker/{uuid}
/sys/fs/cgroup/memory/docker/{uuid}
```

I suspect something related to this bug: https://github.com/opencontainers/runc/issues/931

What's your CoreOS and systemd version?

**Comment by IzabellaRaulin** [comment]
@nanliu, together with @pkudaiyar we confirm that the issue is related with the filepath as You mention.

This filepath can  be different depends on chosen docker [native.cgroupdriver] (https://docs.docker.com/engine/reference/commandline/dockerd/#/options-for-the-runtime) option (which can be set as `cgroupfs` or  `systemd`) and setting of docker [cgroups parents](https://docs.docker.com/engine/reference/commandline/dockerd/#/default-cgroup-parent). 

To simplify, this filepath can look like below:
```
/sys/fs/cgroup/cpuacct/docker/{uuid}
or
/sys/fs/cgroup/cpuacct/system.slice/docker-{uuid}.scope
```

The docker collector in version 3 (current master branch) retrieves data from  `/sys/fs/cgroup/cpuacct/docker/{uuid}`, but there is prepared version 4 (https://github.com/intelsdi-x/snap-plugin-collector-docker/pull/32; unit tests in progress) which makes simply detection which of this two options of possible filepath should be considered as a right path.

To sum up, https://github.com/intelsdi-x/snap-plugin-collector-docker/pull/32 will resolve the issue with running this docker collector on CoreOs.

**Comment by andrzej-k** [comment]
Seems that CoreOS support is there, so I'm closing this issue. Thank you @pkudaiyar for reporting it. In case of any further requests please don't hesitate to open new issue.
